physics/GFS_DCNV_generic.F90

```diff
@@ -63,11 +63,19 @@ subroutine GFS_DCNV_generic_pre_run (im, levs, ldiag3d, qdiag3d, do_cnvgwd, cplc
       endif
 
       if ((ldiag3d.and.qdiag3d) .or. cplchm) then
-        do k=1,levs
-          do i=1,im
-            save_qv(i,k) = gq0_water_vapor(i,k)
+        if(nsamftrac>0) then
+          do n=1,nsamftrac
+            if(n==ntqv .or. dtidx(n+100,index_of_process_dcnv)) then
```

Comment:
This approach assumes that every model stores its tracers in an array FV3 does. This is not interoperable, but maybe it is ok for the time being. I do see the need for looping over tracers and tendencies without knowing what they are (except water vapor), and since we cannot auto-generate the CCPP metadata and Fortran code to automatically adjust to the requested tracers, there is no other easy way for the moment.

Reply:
This is intended to be a medium-term solution, until the CCPP framework is replaced. If the current framework is kept long-term, we can evolve this solution towards a model-independent one. That won't be simple though.